src/api/pipeline/client.ml

```diff
@@ -0,0 +1,84 @@
+open Capnp_rpc_lwt
+
+module State = struct
+  open Raw.Reader.JobInfo.State
+
+  type t = unnamed_union_t
+
+  let pp f = function
+    | NotStarted -> Fmt.string f "not started"
+    | Aborted -> Fmt.string f "aborted"
+    | Failed m -> Fmt.pf f "failed: %s" m
+    | Passed -> Fmt.string f "passed"
+    | Active -> Fmt.string f "active"
+    | Undefined x -> Fmt.pf f "unknown:%d" x
+
+  let from_build_status = function
+    | `Failed -> Failed ""
+    | `Not_started -> NotStarted
+    | `Pending -> Active
+    | `Passed -> Passed
+end
+
+module Build_status = struct
+  include Raw.Reader.BuildStatus
+
+  let pp f = function
+    | NotStarted -> Fmt.string f "not started"
+    | Failed -> Fmt.pf f "failed"
+    | Passed -> Fmt.string f "passed"
+    | Pending -> Fmt.string f "pending"
+    | Undefined x -> Fmt.pf f "unknown:%d" x
+
+  let to_string = function
+  | NotStarted -> "not started"
+  | Failed -> "failed"
+  | Passed -> "passed"
+  | Pending -> "pending"
+  | Undefined _ -> "unknown"
+end
+
+module Project = struct
+  type t = Raw.Client.Project.t Capability.t
+  type project_version = { version : string }
+  type project_status = { version : string; status : Build_status.t }
```

Comment:
We might need to layer a more structured type on top of `version : string` here. Is there a future change that might require an ordering of project versions?

Reply:
Good call - I've switched to using `OpamPackage.Version.t`